classes/items/itembase.cpp

```diff
@@ -18,44 +18,51 @@ void Item::generate()
 
 
 //Constructors / Destructors
-Item::Item()
-{
-    std::string name;
-    unsigned type;
-    unsigned value;
-
-    this->name = ""; //assigns the initial value of name to null
```

Comment:
the constructor should be in this class if you're going to have methods forward declartion and declaration separated.